### DISCUSSION TITLE
expectError() Behavior

### DISCUSSION THREAD

**Discussion by snarkipus** [eclipse-langium/langium#654] [Q&A] [answered]
Not sure if this is an issue or not, but the behavior doesn't quite seem right.  If needed, I can try to come up with a minimal test case.

### Validator Test Case:
```typescript
...
    it('Should detect method invocation on a field', () => {
        const mark = validationResult.document.textDocument.getText().lastIndexOf('(');
        expectError(validationResult, "Method invocation on a field",
        {
            offset: mark,
            length: 2,
        });
    });
...
```
Note: Test passes with length of 2, but fails with length of 1 (off by one in range)

### Code [langium-tests.ts](https://github.com/langium/langium/blob/4af91a15ff4275b434281e90330d5f1ebaacc0a5/packages/langium/src/test/langium-test.ts#L318-L324)
```typescript
...
    if ('offset' in options) {
        const outer = {
            start: validationResult.document.textDocument.positionAt(options.offset),
            end: validationResult.document.textDocument.positionAt(options.offset + options.length - 1)
        };
        filters.push(d => isRangeEqual(outer, d.range));
    }
...
```
Note: wondering if the `end` decrement `options.offset + options.length - 1` is correct?
### Validator in action:
<img width="742" alt="image" src="https://user-images.githubusercontent.com/3688989/187049253-9a489fea-fba8-4335-b425-c1e43228e9fb.png">

I don't believe that this would have been caught in the tests since it's uncovered in the test report:
<img width="626" alt="image" src="https://user-images.githubusercontent.com/3688989/187049493-9bfcdaf8-502e-4b22-8ec6-318d89878474.png">

**Comment by msujew** [comment, marked as answer]
I created an issue from this, since this is definitely not intended to work that way https://github.com/langium/langium/issues/655.

**Reply by snarkipus** [reply]
Thanks @msujew. I updated the issue you created with a minimal example and a link to some example code. Maybe I'll stumble across a fix - if so I'll submit a PR.
